### PR TITLE
moving side-nav, router, and shared vuex store to core

### PR DESCRIPTION
**Pull request by indirectlylit**

### CODE REVIEW

File: kolibri/core/assets/src/vue/side-nav/learn.svg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<svg fill="#996189" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
```
Comment (rtibbles) [inline kolibri/core/assets/src/vue/side-nav/learn.svg]: This is feeling somewhat unpluggable right now. OK for MVP, but we will want to revisit how these assets could be pluggable.

Comment (indirectlylit) [inline kolibri/core/assets/src/vue/side-nav/learn.svg]: yeah - this PR is going to be full of hacks :(

File: kolibri/core/kolibri_plugin.py [hunk 2]
```diff
@@ -15,17 +12,11 @@ class KolibriCore(KolibriPluginBase):
     pass
 
 
-class MainNavigationItem(hooks.NavigationHook):
-    label = _("Do nothing")
-    url = '#'
-
-
```
Comment (jamalex) [inline kolibri/core/kolibri_plugin.py:22]: I added a "Technical debt (clean up post-MVP)" list on the tasks board for tracking things we'll want to revisit later, and added a card for the pluggable nav.

File: kolibri/core/assets/src/vue/side-nav/learn.svg (hunk 1 above)
Comment (jamalex) [inline kolibri/core/assets/src/vue/side-nav/learn.svg]: Added a "technical debt" card to track this for later:
https://trello.com/b/bjHannKQ/kolibri-tasks

File: kolibri/plugins/learn/assets/src/vue/side-nav/nav-item.vue [hunk 3]
```diff
@@ -0,0 +1,82 @@
+<template>
+
+  <div class='link-wrapper'>
+    <a class='link' v-link="vlink" @click='closeSearch' :class="{active: active}">
```
Comment (jamalex) [inline kolibri/plugins/learn/assets/src/vue/side-nav/nav-item.vue]: Having `closeSearch` here feels a bit weird/backwards -- what if there's other stuff that should happen when top-level nav is changed? Might make more sense to have the apps/plugins themselves listen for nav changes (some sort of "page exit" event?) and do whatever they want at that point.

Comment (jamalex) [inline kolibri/plugins/learn/assets/src/vue/side-nav/nav-item.vue]: Ah, I see the [comment below](https://github.com/learningequality/kolibri/pull/319/commits/a4aa4b927317b5e9fd6f66ce90582b4f759796fe#diff-9a18a9cca259b56041fa61a01a4fa2bfR34). Perhaps `@click` could be bound to a more generic "top-level-nav-item-clicked" type event, and the apps can listen for that as desired.

**Comment by jamalex** [comment]
Because the PR is so big, I'm going through commit-by-commit. That led to a comment just now being already marked as "outdated", so I'm posting a reference to it here:
https://github.com/learningequality/kolibri/pull/319/commits/a4aa4b927317b5e9fd6f66ce90582b4f759796fe#r73554714

File: kolibri/plugins/learn/assets/src/vue/main-nav/index.vue [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      },
+      learnActive() {
+        return this.pageMode === constants.PageModes.LEARN;
+      },
+      exploreLink() {
+        return { name: constants.PageNames.EXPLORE_ROOT };
+      },
+      exploreActive() {
+        return this.pageMode === constants.PageModes.EXPLORE;
+      },
```
Comment (jamalex) [inline kolibri/plugins/learn/assets/src/vue/main-nav/index.vue:51]: This duplication seems like something that could be done more systematically; adding a link into the cleanup cards for later.

File: kolibri/plugins/management/assets/src/state/constants.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+
+// a name for every URL pattern
+const PageNames = {
+  USER_PAGE: 'USER_PAGE',
+  CONTENT_PAGE: 'CONTENT_PAGE',
```
Comment (jamalex) [inline kolibri/plugins/management/assets/src/state/constants.js]: `CONTENT_PAGE` is under `management` plugin?

File: kolibri/plugins/management/assets/src/vue/top-nav/index.vue [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    <a v-link="contentLink" :class="{active: contentActive}">Content</a>
+  </div>
+
+</template>
+
+
+<script>
+
+  const constants = require('../../state/constants');
+  console.log('CO>>SAD', constants);
```
Comment (jamalex) [inline kolibri/plugins/management/assets/src/vue/top-nav/index.vue]: Perhaps remove if it was just for debugging.

**Comment by jamalex** [comment]
Cool, I've finished a pass through, and added a couple of comments, basically to highlight spots for us to revisit later. Let's get this merged after standup.

File: frontend_build/src/read_bundle_plugins.js [hunk 7]
```diff
@@ -24,8 +24,6 @@ var libs = function(kolibri_name) {
     'logging': kolibri_name + '.lib.logging',
     'vue': kolibri_name + '.lib.vue',
     'kolibri': kolibri_name,
-    'core-base': kolibri_name + '.lib.coreBase',
```
Comment (rtibbles) [inline frontend_build/src/read_bundle_plugins.js:27]: What is the purpose of these changes? This will result in the core-base and content-renderer code being seperately bundled into every app that uses it, rather than being in the core-app.

File: kolibri/core/assets/src/core-app/constructor.js [hunk 8]
```diff
@@ -33,12 +33,9 @@ const publicMethods = [
  */
 function Lib() {
   // libraries
-  this.logging = require('./logging');
+  this.logging = require('../logging');
   this.vue = vue;
   this.vuex = vuex;
-  // views
-  this.coreBase = require('./core-base');
```
Comment (rtibbles) [inline kolibri/core/assets/src/core-app/constructor.js:40]: Ditto here.

**Comment by rtibbles** [comment]
Only skimmed - mostly makes sense to me, am concerned about the point I raised above about the removing of the externals though, would want to see that resolved/addressed before merge.

File: kolibri/plugins/management/assets/src/state/constants.js (hunk 5 above)
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/state/constants.js]: content import - maybe I should call it channels?

File: kolibri/plugins/management/assets/src/vue/top-nav/index.vue (hunk 6 above)
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/top-nav/index.vue]: funny how easy it is to type the word 'sad' by mashing keys

File: frontend_build/src/read_bundle_plugins.js (hunk 7 above)
Comment (indirectlylit) [inline frontend_build/src/read_bundle_plugins.js:27]: ok - I'll re-instate them.

I still don't fully understand how these functions and was just trying to clean up. It's not clear to me how these interact with webpack.config.base.js' `resolve` -> `alias`, which itself is a bit hard to maintain

Comment (rtibbles) [inline frontend_build/src/read_bundle_plugins.js:27]: The alias stuff is necessary to allow any test code to build properly - these ones make the mapping for production/dev environments to be able to reach into the core app and access code.

I agree it could all be managed in a neater way, just still not sure how currently!

Comment (indirectlylit) [inline frontend_build/src/read_bundle_plugins.js:27]: Is there a chance that a lot of code that runs in my dev environment is broken in production because I haven't been adding core libs to the `Lib` object?

Comment (rtibbles) [inline frontend_build/src/read_bundle_plugins.js:27]: No, dev and production work identically, the only danger is of code duplication, where code that should only be in core gets repeated across other modules.
